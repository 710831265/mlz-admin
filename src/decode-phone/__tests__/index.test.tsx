import React from 'react';
import { testMount, testSnapshot, getNodeRef } from '../../../tests';
import DecodePhone from '..';
import { mount } from 'enzyme';

describe('ðŸ§ª DecodePhone', () => {
  testMount(DecodePhone);
  testSnapshot(DecodePhone);

  it('snapshot', () => {
    const wrapper = mount(
      <DecodePhone params="jZgOvsexafxJUlU3WHaMfA==">
        <span id="phone">155****1234</span>
      </DecodePhone>,
    );
    const div = wrapper.find('#phone').at(0);
    div.simulate('click');
    expect(getNodeRef(wrapper, '.ant-tooltip-inner').innerHTML).not.toEqual(/åŠ è½½ä¸­/);
  });

  it('second click will not do request', () => {
    const theOnChangeInRC = jest.spyOn(console, 'log');
    const wrapper = mount(
      <DecodePhone params="jZgOvsexafxJUlU3WHaMfA==">
        <span id="phone">155****1234</span>
      </DecodePhone>,
    );
    const div = wrapper.find('#phone').at(0);
    div.simulate('click');
    expect(theOnChangeInRC).toHaveBeenCalledTimes(1);
    div.simulate('click');
    div.simulate('click');
    expect(theOnChangeInRC).toHaveBeenCalledTimes(1);
  });
});
