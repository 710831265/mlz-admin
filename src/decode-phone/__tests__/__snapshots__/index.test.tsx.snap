// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª DecodePhone Snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "ant-tooltip-open",
      "id": "phone",
    },
    "children": Array [
      Object {
        "data": "155****1234",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "span",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "class": "ant-tooltip",
            "style": "opacity: 0; pointer-events: none;",
          },
          "children": Array [
            Object {
              "attribs": Object {
                "class": "ant-tooltip-content",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "ant-tooltip-arrow",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "ant-tooltip-arrow-content",
                      },
                      "children": Array [],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": Object {
                    "attribs": Object {
                      "class": "ant-tooltip-inner",
                      "role": "tooltip",
                    },
                    "children": Array [
                      Object {
                        "data": "åŠ è½½ä¸­",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                      "role": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                      "role": undefined,
                    },
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                Object {
                  "attribs": Object {
                    "class": "ant-tooltip-inner",
                    "role": "tooltip",
                  },
                  "children": Array [
                    Object {
                      "data": "åŠ è½½ä¸­",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "ant-tooltip-arrow",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "ant-tooltip-arrow-content",
                        },
                        "children": Array [],
                        "name": "span",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                    ],
                    "name": "div",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "role": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "role": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "style": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "style": undefined,
          },
        },
      ],
      "name": "div",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": [Circular],
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
        Object {
          "attribs": Object {},
          "children": Array [
            Object {
              "attribs": Object {
                "class": "ant-tooltip",
                "style": "opacity: 0; pointer-events: none;",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "ant-tooltip-content",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "ant-tooltip-arrow",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "ant-tooltip-arrow-content",
                          },
                          "children": Array [],
                          "name": "span",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "class": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "class": undefined,
                          },
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "class": "ant-tooltip-inner",
                          "role": "tooltip",
                        },
                        "children": Array [
                          Object {
                            "data": "åŠ è½½ä¸­",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                          "role": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                          "role": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "class": "ant-tooltip-inner",
                        "role": "tooltip",
                      },
                      "children": Array [
                        Object {
                          "data": "åŠ è½½ä¸­",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "div",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "class": "ant-tooltip-arrow",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "ant-tooltip-arrow-content",
                            },
                            "children": Array [],
                            "name": "span",
                            "namespace": "http://www.w3.org/1999/xhtml",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                            "x-attribsNamespace": Object {
                              "class": undefined,
                            },
                            "x-attribsPrefix": Object {
                              "class": undefined,
                            },
                          },
                        ],
                        "name": "div",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "class": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "class": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                        "role": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                        "role": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
              ],
              "name": "div",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
                "style": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
                "style": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": null,
          "prev": [Circular],
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "id": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "id": undefined,
    },
  },
  "1": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "ant-tooltip",
          "style": "opacity: 0; pointer-events: none;",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ant-tooltip-content",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "ant-tooltip-arrow",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "ant-tooltip-arrow-content",
                    },
                    "children": Array [],
                    "name": "span",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "class": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "class": undefined,
                    },
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "ant-tooltip-inner",
                    "role": "tooltip",
                  },
                  "children": Array [
                    Object {
                      "data": "åŠ è½½ä¸­",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "role": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "role": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "ant-tooltip-inner",
                  "role": "tooltip",
                },
                "children": Array [
                  Object {
                    "data": "åŠ è½½ä¸­",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "div",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "ant-tooltip-arrow",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "ant-tooltip-arrow-content",
                      },
                      "children": Array [],
                      "name": "span",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "class": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "class": undefined,
                      },
                    },
                  ],
                  "name": "div",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "role": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "role": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
          "style": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
          "style": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": Object {
      "attribs": Object {
        "class": "ant-tooltip-open",
        "id": "phone",
      },
      "children": Array [
        Object {
          "data": "155****1234",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "text",
        },
      ],
      "name": "span",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": [Circular],
      "parent": null,
      "prev": null,
      "root": Object {
        "attribs": Object {},
        "children": Array [
          [Circular],
          [Circular],
        ],
        "name": "root",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": null,
        "prev": null,
        "type": "root",
        "x-attribsNamespace": Object {},
        "x-attribsPrefix": Object {},
      },
      "type": "tag",
      "x-attribsNamespace": Object {
        "class": undefined,
        "id": undefined,
      },
      "x-attribsPrefix": Object {
        "class": undefined,
        "id": undefined,
      },
    },
    "root": Object {
      "attribs": Object {},
      "children": Array [
        Object {
          "attribs": Object {
            "class": "ant-tooltip-open",
            "id": "phone",
          },
          "children": Array [
            Object {
              "data": "155****1234",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": null,
          "prev": null,
          "root": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
            "id": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
            "id": undefined,
          },
        },
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 2,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;

exports[`ðŸ§ª DecodePhone snapshot test snapshot comparing 1`] = `"<span></span>"`;
